src/test/java/com/scouts/kitchenplanerbackend/ProjectRepositoryTest.java

```diff
@@ -0,0 +1,158 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+package com.scouts.kitchenplanerbackend;
+
+import com.scouts.kitchenplanerbackend.entities.UserEntity;
+import com.scouts.kitchenplanerbackend.entities.projects.ProjectEntity;
+import com.scouts.kitchenplanerbackend.entities.projects.ProjectStubDTO;
+import com.scouts.kitchenplanerbackend.repositories.UserRepository;
+import com.scouts.kitchenplanerbackend.repositories.projects.ProjectRepository;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+
+import java.util.Collection;
+import java.util.Date;
+
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@DataJpaTest
+public class ProjectRepositoryTest {
+
+    @Autowired
+    private ProjectRepository projectRepo;
+
+    @Autowired
+    private UserRepository userRepo;
+
+
+    @BeforeEach
+    void setUp() {
+        ProjectEntity project = new ProjectEntity();
+        project.setName("Test Project");
+        projectRepo.save(project);
+
+        assertEquals(1, projectRepo.findAll().size());
+        assertEquals("Test Project", projectRepo.findAll().get(0).getName());
+
+    }
+
+
+    @Test
+    void findCorrectAmountOfParticipants() {
+        UserEntity user1 = new UserEntity("user1");
+        UserEntity user2 = new UserEntity("user2");
+
+        user1 = userRepo.save(user1);
+        user2 = userRepo.save(user2);
+
+
+        assertEquals(2, userRepo.findAll().size());
+        long userId1 = user1.getId();
+        long userId2 = user2.getId();
+
+        ProjectEntity project = projectRepo.findAll().get(0);
+        long projectID = project.getId();
+
+
+        assertFalse(projectRepo.existsByParticipants_IdAndId(projectID, userId1));
+        project.getParticipants().add(user1);
+        project = projectRepo.save(project);
+        assertTrue(projectRepo.existsByParticipants_IdAndId(projectID, userId1));
+
+
+        assertEquals(1, projectRepo.countMembersById(projectID));
+        Collection<UserEntity> u = projectRepo.findParticipantsById(projectID);
+        assertEquals(1, u.size());
+        assertEquals(user1, u.toArray()[0]);
+
+        project.getParticipants().add(user2);
+        project = projectRepo.save(project);
+        assertEquals(2, projectRepo.countMembersById(projectID));
+        u = projectRepo.findParticipantsById(projectID);
+        assertEquals(2, u.size());
+        assertEquals(user1, u.toArray()[0]);
+        assertEquals(user2, u.toArray()[1]);
```

Comment:
This is probably non-deterministic. Change to check that both users are contained in the collection but don't rely on the order of the collection.